spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendResponseFilter.java

```diff
@@ -172,27 +173,25 @@ else if (context.getResponseGZipped() && isGzipRequested) {
 		}
 		finally {
 			try {
-				if (is != null) {
```

Comment:
IIRC correctly closing the InputStream simply had no effect. The InputStream instance was some kind of wrapped stream, wich tried to reach the end of the stream on closing... which resulted in waiting forever on infinite chunked http response.
IMHO the InputStream shloudn't be necessary... If you insist on closing the stream it must be done after closing the response (but will result in an IOException in the mentioned edge case on trying to reach the end, since the underlying HttpResponse is closed).